cfme/fixtures/vm.py

```diff
@@ -1,44 +1,76 @@
+        )
+    )
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_collection_finish(session):
+    """This hook will call the process_json_data and cause _docstrings_cache
+    to be populated so that we can access required information in the _create_vm."""
+    if not session.config.getoption('--no-assignee-vm-name'):
+        process_json_data(session, session.items, all_items=True)
 
 
 def _create_vm(request, template, provider, vm_name):
+    if not request.config.getoption('--no-assignee-vm-name'):
+        if isinstance(request.node, pytest.Function):
+            assignee = get_parsed_docstring(request.node,
+                request.session._docstrings_cache).get('assignee', '')
+        else:
+            # Fetch list of tests in the module object
+            test_list = [
+                item
+                for item in dir(request.module)
+                if item.startswith('test_') and not ('test_requirements' == item)
+            ]
+            # Find out assignee for each test in test_list
+            assignee_list = list()
+            for test in test_list:
+                nodeid = f'{request.node.fspath.strpath}::{test}'
+                try:
+                    assignee_list.append(request.session._docstrings_cache[nodeid]['assignee'])
+                except KeyError:
+                    continue
+            # If all tests have same assignee, set length will be 1, else set assignee='module'
+            assignee = assignee_list[0] if len(set(assignee_list)) == 1 else 'module'
+        vm_name = f'{vm_name}-{assignee}'
     collection = provider.appliance.provider_based_collection(provider)
-    vm_obj = collection.instantiate(vm_name, provider, template_name=template.name)
-    vm_obj.create_on_provider(allow_skip="default")
-    vm_obj.mgmt.ensure_state(VmState.RUNNING)
-    # In order to have seamless SSH connection
-
-    vm_ip, _ = wait_for(
-        lambda: vm_obj.ip_address,
-        num_sec=300,
-        delay=5,
-        fail_condition=None,
-        message="wait for testing VM pingable IP address."
-    )
-    wait_for(
-        net_check,
-        func_args=[ports.SSH, vm_ip],
-        func_kwargs={"force": True},
-        num_sec=300,
-        delay=5,
-        message="testing VM's SSH available")
-    if not vm_obj.exists:
-        provider.refresh_provider_relationships()
-        vm_obj.wait_to_appear()
+    try:
+        vm_obj = collection.instantiate(
+            vm_name, provider,
+            template_name=provider.data['templates'][template].name
+        )
+    except (KeyError, AttributeError):
+        pytest.skip('No appropriate template was passed to the fixture.')
+    vm_obj.create_on_provider(allow_skip='default')
 
     @request.addfinalizer
     def _cleanup():
         vm_obj.cleanup_on_provider()
         provider.refresh_provider_relationships()
 
+    vm_obj.mgmt.ensure_state(VmState.RUNNING)
```

Comment:
Why remove the net_check and ip_address wait_for calls in the fixture currently?

Reply:
As discussed offline, not all templates have ability to share the Pingable IPs as sometimes some of them do not have Guest tools installed and not all the tests need these either, so having a check to test IP etc can cause failure even if test case calling this fixture doesn't care for IP.